Persephone/AppDelegate.swift

```diff
@@ -10,15 +10,41 @@ import Cocoa
 
 @NSApplicationMain
 class AppDelegate: NSObject, NSApplicationDelegate {
+  var preferences = Preferences()
+
   static let mpdClient = MPDClient(
     withDelegate: NotificationsController()
   )
  
   func applicationDidFinishLaunching(_ aNotification: Notification) {
-    AppDelegate.mpdClient.connect()
+    connect()
+
+    UserDefaults.standard.addObserver(self, forKeyPath: "mpdHost", options: .new, context: nil)
+    UserDefaults.standard.addObserver(self, forKeyPath: "mpdPort", options: .new, context: nil)
```

Comment:
Is this maybe leaking an implementation detail of `Preferences`? Would it make sense to add an observation helper method on `Preferences`? Would it make sense to have `Preferences` be a singleton that posts a notification when either preference changes?

Reply:
I like the observer helper method idea. I'm going to do that for now, but I may need the singleton pattern later.